plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -265,29 +472,46 @@ public JSONArray getPluginDependencyJsonArray(Plugin plugin, JSONObject ucJson)
      *
      * @param plugin
      */
-    public void resolveDependenciesFromManifest(Plugin plugin) {
-        String dependencyString = getAttributefromManifest(plugin.getFile(), "Plugin-Dependencies");
-        String[] dependencies = dependencyString.split(",");
+    public List<Plugin> resolveDependenciesFromManifest(Plugin plugin) {
         List<Plugin> dependentPlugins = new ArrayList<>();
 
-        System.out.println(plugin.getName() + " depends on: ");
-
-        boolean isPluginOptional = false;
+        try {
+            File tempFile = Files.createTempFile(plugin.getName(), ".jpi").toFile();
+            if (!downloadPlugin(plugin, tempFile)) {
+                System.out.println("Unable to resolve dependencies");
+                Files.delete(tempFile.toPath());
+                return dependentPlugins;
+            }
+            String dependencyString = getAttributefromManifest(tempFile, "Plugin-Dependencies");
+            String[] dependencies = dependencyString.split(",");
 
-        for (String dependency : dependencies) {
-            if (dependency.contains("resolution:=optional")) {
-                dependency = dependency.split(";")[0];
-                isPluginOptional = true;
+            if (isVerbose) {
+                System.out.println("\n" + plugin.getName() + " depends on: ");
             }
-            String[] pluginInfo = dependency.split(":");
-            String pluginName = pluginInfo[0];
-            String pluginVersion = pluginInfo[1];
-            Plugin dependentPlugin = new Plugin(pluginName, pluginVersion, isPluginOptional);
-            dependentPlugins.add(dependentPlugin);
 
-            System.out.println(pluginName + ": " + pluginVersion);
+            boolean isPluginOptional = false;
+
+            for (String dependency : dependencies) {
+                if (dependency.contains("resolution:=optional")) {
+                    dependency = dependency.split(";")[0];
+                    isPluginOptional = true;
+                }
+                String[] pluginInfo = dependency.split(":");
+                String pluginName = pluginInfo[0];
+                String pluginVersion = pluginInfo[1];
+                Plugin dependentPlugin = new Plugin(pluginName, pluginVersion, isPluginOptional);
+                dependentPlugins.add(dependentPlugin);
+
+                if (isVerbose) {
+                    System.out.println(pluginName + ": " + pluginVersion);
+                }
+            }
+            Files.delete(tempFile.toPath());
+            return dependentPlugins;
+        } catch (IOException e) {
+            System.out.println("Unable to resolve dependencies");
```

Comment:
You might want to log this error, and minimally what plugin you were processing when the error occurred.

Reply:
If there are any errors, I am planning on dealing with these by keeping track of the initial plugins in the plugin directory and then restoring those and deleting anything that was downloaded. I haven't gotten to it yet though, so Idk if we should try to include those changes in the upcoming alpha or not and maybe I should just implement something else in the meantime.